src/parser.rs

```diff
@@ -109,6 +109,10 @@ pub(crate) fn parse_transport_type(input: &str) -> Result<TransportLayer, NixUri
     TryInto::<TransportLayer>::try_into(input)
 }
 
+pub(crate) fn parse_sep(input: &str) -> IResult<&str, &str> {
+    tag("://")(input)
+}
+
```

Comment:
Useful for when we start passing up usefull errors. Failing here will easily be mapped to an error that says "expected `://` but got `...`